java/src/main/resources/web/connect-e.js

```diff
@@ -0,0 +1,285 @@
+  if (errors && errors.length) {
+    const list = document.createElement('ul');
+    errors.forEach(function (error) {
+      const item = document.createElement('li');
+      item.innerText = error.message;
+      list.appendChild(item);
+    });
+    errorsDiv.appendChild(list);
+  }
+}
+
+const styles = {
+  base: {
+    default: {
+      color: 'black',
+      textDecoration: 'none',
+      fontFamily: "'Do Hyeon', sans-serif",
+      boxSizing: 'border-box',
+      padding: '.375rem .75rem',
+      boxShadow: 'none',
+      fontSize: '1rem',
+      borderRadius: '.25rem',
+      lineHeight: '1.5',
+      backgroundColor: '#fff',
+    },
+    focus: {
+      color: '#495057',
+      borderColor: '#80bdff',
+    },
+    error: {
+      color: '#B00',
+      borderColor: '#B00',
+    },
+    valid: {
+      color: 'green',
+      borderColor: 'green',
+    },
+    label: {
+      display: 'none',
+    },
+  },
+  cv2: {
+    container: {
+      width: '25%',
+      float: 'left',
+      boxSizing: 'border-box',
+    },
+    default: {
+      borderRadius: '0 .25rem .25rem 0',
+    },
+  },
+  expiryDate: {
+    container: {
+      width: '25%',
+      float: 'left',
+      borderRadius: '0rem',
+    },
+    default: {
+      borderRadius: '0',
+      borderRight: 'none',
+    },
+  },
+
+  cardNumber: {
+    container: {
+      width: '50%',
+      float: 'left',
+    },
+    default: {
+      borderRadius: '.25rem 0 0 .25rem',
+      borderRight: 'none',
+    },
+  },
+};
+
+const form = document.getElementById('txn-type-select-form');
+
+const buildRefundButton = (xref, atData) => {
+  // after sale, process as REFUND
+  atData.transactionType = "REFUND";
+  atData.crossReference = xref;
+
+  let refundButton = document.createElement("button");
+  refundButton.innerHTML = "Refund";
+  refundButton.onclick = function() {
+    refundButton.disabled = true;
+    console.log("clicked refund button");
+
+    // create a new access token with the updated transaction type
+    fetch('/api/access-tokens',  {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(atData),
+    }).then((accessTokensResponse) => {
+      return accessTokensResponse.json();
+    })
+    .then((accessTokensResponse) => {
+      let xrefUrl = `/api/cross-reference-payments/${accessTokensResponse.id}`;
+      fetch(xrefUrl, {
+        method: 'POST',
+        body: JSON.stringify({ crossReference: xref })
+      })
+      .then((xrefResponse) => {
+        return xrefResponse.json();
+      })
+      .then((xrefResponse) => {
+        // update refund div with response status
+        document.getElementById("refund").innerHTML = JSON.stringify(xrefResponse);
+        return xrefResponse;
+      })
+      .catch((error) => {
+        console.error(`${xrefUrl} - error`, JSON.stringify(error));
+        throw error;
+      });
+    });
+  }
+  refundButton.classList.add("btn-primary"); // btn pull-right
```

Comment:
Any reason the button is created in JS rather then just binding events to one rendered in HTML?

Reply:
Nope.  Will adjust before I push!